src/view/display-lore.cpp

```diff
@@ -311,14 +311,14 @@ void display_monster_move(lore_type *lore_ptr)
 
 void display_random_move(lore_type *lore_ptr)
 {
-    if (((lore_ptr->flags1 & RF1_RAND_50) == 0) && ((lore_ptr->flags1 & RF1_RAND_25) == 0))
+    if (lore_ptr->race_behavior_flags.has_not(MonraceBehaviorType::RAND_MOVE_50) && lore_ptr->race_behavior_flags.has_not(MonraceBehaviorType::RAND_MOVE_25))
```

Comment:
「指定されたフラグを全て持っていない」なので、以下と等価です
```
!lore_ptr->race_behavior_flags.has_all_of( { MonraceBehaviorType::RAND_MOVE_50, MonraceBehaviorType::RAND_MOVE_25 } )
```

Reply:
!～.has_all_of よりも ～.has_none_of を使うべきですね。
```suggestion
   lore_ptr->race_behavior_flags.has_none_of( { MonraceBehaviorType::RAND_MOVE_50, MonraceBehaviorType::RAND_MOVE_25 } )
```

Reply:
確かにその通りですね
FlagGroupは本当に便利ですな

Reply:
ついでにいうと元の条件と同義なのは !～.has_any_of ですね。（実際FlagGroupクラスのhas_none_ofは!has_any_ofで実装されています）

Reply:
has_none_ofでまとめました